ms-common-auth/src/main/java/net/trajano/ms/authz/TokenResource.java

```diff
@@ -109,64 +97,77 @@ public OAuthTokenResponse dispatch(
             return handleJwtAssertionGrant(assertion, clientId);
 
         } else {
-            throw OAuthTokenResponse.badRequest("invalid_grant", "Invalid grant");
+            throw OAuthTokenResponse.badRequest(ErrorCodes.UNSUPPORT_GRANT_TYPE, "Invalid grant");
         }
 
     }
 
+    @Autowired
+    private CryptoOps cryptoOps;
+
     private IdTokenResponse handleAuthorizationCodeGrant(final String accessToken,
-        @NotNull final String clientId) {
+        final String clientId) {
 
         if (accessToken == null) {
             throw OAuthTokenResponse.badRequest("invalid_request", "Missing access token");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "invalid_request" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)